# Week 8 Activity test

# 1. Carry out descriptive analytics on the dataset

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
data=pd.read_csv("E:\Python Programming\Assignments\Week_8_Q&A_dataset - Sheet1.csv")

In [3]:
data.head()

Gender Test preparation  Total Marks
0    male             none           14
1  female             none           28
2  female             none           18
3  female             none           48
4  female             none           21

In [4]:
data.dtypes

Gender              object
Test preparation    object
Total Marks          int64
dtype: object

Datatype of column need to be changed 

In [5]:
data['Gender']=data['Gender'].astype('str')

In [6]:
(data['Gender']=='female').sum()

16

In [12]:
data['Total Marks'].groupby(data['Gender']).sum()

Gender
female    540
male      365
Name: Total Marks, dtype: int64

In [13]:
data['Total Marks'].groupby(data['Gender']).mean()

Gender
female    33.750000
male      30.416667
Name: Total Marks, dtype: float64

### As part iof initial analysis,looking at the sum and mean of Total Marks genderwise, 
### female students are performing better than boys.



# 2. Check whether female and male students scored the same marks.


## Two Tailed t-test

### Null Hypothesis :- Female and male students scored the same marks 
### Alternate Hypothesis :- Female and male students scored different marks

In [15]:
data['Total Marks'].groupby(data.Gender).mean()

Gender
female    33.750000
male      30.416667
Name: Total Marks, dtype: float64

In [19]:
female_marks=data[data['Gender']=='female']

In [20]:
male_marks=data[data['Gender']=='male']

In [21]:
from scipy.stats import ttest_ind

In [23]:
ttest_ind(female_marks['Total Marks'],male_marks['Total Marks'],equal_var=False)

Ttest_indResult(statistic=0.6998399299219519, pvalue=0.4906112127110316)

### From the Ttest performed, pvalue=0.4906112127110316 > 0.05 and hence we cannot reject the null Hypothesis
### i.e, we conclude the test result as:-
### Female and male students scored the same marks

# 3. Whether test preparation helps the students?

## One tailed Two sample t-test

### Null Hypothesis: Students completed test preparation has same marks as those who didn't.
### Alternate Hypothesis: Students completed test preparation  have higher marks than those who didn't.

In [27]:
data.head(10)

Gender Test preparation  Total Marks
0    male             none           14
1  female             none           28
2  female             none           18
3  female             none           48
4  female             none           21
5  female        completed           40
6    male             none           30
7  female             none           12
8    male             none           18
9    male             none           24

In [28]:
tp_done=data[data['Test preparation']=='completed']

In [29]:
notp_done=data[data['Test preparation']=='none']

In [40]:
tp_done.mean()

Total Marks    44.571429
dtype: float64

In [41]:
notp_done.mean()

Total Marks    28.238095
dtype: float64

In [49]:
ttest_ind(tp_done['Total Marks'],notp_done['Total Marks'])

Ttest_indResult(statistic=3.6143637527769217, pvalue=0.0012671292457320106)

### Here pvalue=0.0012671292457320106 < 0.05 and hence we reject the null hypothesis and accept the 
### Alternate hypothesis
### i.e we conclude this test as:-  Students completed test preparation has higher marks than those who didn't.
### i.e Test preparation has an impact on the marks scored by Students.